fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/FDBLuceneMapQueryTest.java

```diff
@@ -0,0 +1,461 @@
+                    .setGroup(i % 2)
+                    .build()
+            )
+            .collect(Collectors.toList());
+
+
+    private ExecutorService executorService = null;
+
+    @Override
+    public void setupPlanner(@Nullable PlannableIndexTypes indexTypes) {
+        if (indexTypes == null) {
+            indexTypes = new PlannableIndexTypes(
+                    Sets.newHashSet(IndexTypes.VALUE, IndexTypes.VERSION),
+                    Sets.newHashSet(IndexTypes.RANK, IndexTypes.TIME_WINDOW_LEADERBOARD),
+                    Sets.newHashSet(IndexTypes.TEXT),
+                    Sets.newHashSet(LuceneIndexTypes.LUCENE)
+            );
+        }
+        planner = new LucenePlanner(recordStore.getRecordMetaData(), recordStore.getRecordStoreState(), indexTypes, recordStore.getTimer());
+    }
+
+    @Override
+    protected RecordLayerPropertyStorage.Builder addDefaultProps(final RecordLayerPropertyStorage.Builder props) {
+        return super.addDefaultProps(props)
+                .addProp(LuceneRecordContextProperties.LUCENE_EXECUTOR_SERVICE, (Supplier<ExecutorService>)() -> executorService);
+    }
+
+    protected void openRecordStore(FDBRecordContext context) {
+        openRecordStore(context, md -> {
+        }, SIMPLE_TEXT_SUFFIXES);
+    }
+
+    protected void openRecordStore(FDBRecordContext context, RecordMetaDataHook hook, Index simpleDocIndex) {
+        RecordMetaDataBuilder metaDataBuilder = RecordMetaData.newBuilder().setRecords(TestRecordsTextProto.getDescriptor());
+        metaDataBuilder.getRecordType(TextIndexTestUtils.COMPLEX_DOC).setPrimaryKey(concatenateFields("group", "doc_id"));
+        if (simpleDocIndex != null) {
+            metaDataBuilder.removeIndex("SimpleDocument$text");
+            metaDataBuilder.addIndex(TextIndexTestUtils.SIMPLE_DOC, simpleDocIndex);
+        }
+        metaDataBuilder.addIndex(MAP_DOC, MAP_STRING_2_LONG_LUCENE_INDEX);
+        metaDataBuilder.addIndex(MAP_DOC, MAP_STRING_WRAPPER_TO_LONG_LUCENE_INDEX);
+        metaDataBuilder.addIndex(MAP_DOC, MAP_STRING_2_LONG_WRAPPER_LUCENE_INDEX);
+        metaDataBuilder.addIndex(MAP_DOC, MAP_STRING_2_INT_LUCENE_INDEX);
+        metaDataBuilder.addIndex(MAP_DOC, MAP_STRING_2_DOUBLE_LUCENE_INDEX);
+        hook.apply(metaDataBuilder);
+        recordStore = getStoreBuilder(context, metaDataBuilder.getRecordMetaData())
+                .setSerializer(TextIndexTestUtils.COMPRESSING_SERIALIZER)
+                .createOrOpen();
+        setupPlanner(null);
+    }
+
+    private void initializeNested() {
+        try (FDBRecordContext context = openContext()) {
+            openRecordStore(context);
+            mapDocuments.forEach(recordStore::saveRecord);
+            commit(context);
+        }
+    }
+
+    @ParameterizedTest
+    @CsvSource({
+            "d, true",
+            "Blah, false",
+            "c, false"
+    })
+    void mapStringToLong(String value, boolean found) throws Exception {
```

Comment:
All of these tests are almost exactly the same, would it benefit from having a single `mapKeyQuery` that takes a `@MethodSource`, something like:
```java
    public static Stream<Arguments> mapKeyQuery() {
        return Stream.of(
                Stream.of(Pair.of("d", true),
                                Pair.of("Blah", false),
                                Pair.of("c", false))
                        .map(pair ->
                                Arguments.of(Query.field("stringToLongMap").matches(Query.field("values").oneOfThem().matches(Query.field("key").equalsValue(pair.getKey()))), pair.getRight())),
                Stream.of(Pair.of("c", true),
                                Pair.of("Blah", false),
                                Pair.of("d", false))
                        .map(pair ->
                                Arguments.of(Query.field("stringToLongWrapperMap").matches(Query.field("values").oneOfThem().matches(Query.field("key").equalsValue(pair.getKey()))), pair.getRight()))
        ).flatMap(Function.identity());
    }
```

You might also be able to make it a little more interesting, and cover both the parameterized and literal versions.
 Either that, or maybe just a helper method that takes a couple arguments from here.

Probably not too important either way....